$OpenBSD: patch-inc_My_Build_Any_wx_config_Bakefile_pm,v 1.1.1.1 2009/09/04 16:05:55 landry Exp $
use glob() to find all matching $lib.so.*, and take the last value
--- inc/My/Build/Any_wx_config_Bakefile.pm.orig	Sun Aug  9 13:13:40 2009
+++ inc/My/Build/Any_wx_config_Bakefile.pm	Fri Aug 28 15:29:47 2009
@@ -3,7 +3,6 @@ package My::Build::Any_wx_config_Bakefile;
 use strict;
 our @ISA = qw(My::Build::Any_wx_config::Base);
 use Config;
-
 sub awx_wx_config_data {
     my $self = shift;
     return $self->{awx_data} if $self->{awx_data};
@@ -24,7 +23,7 @@ sub awx_wx_config_data {
     my $libraries = $self->_call_wx_config( $arg );
 
     foreach my $lib ( grep { m/\-lwx/ } split ' ', $libraries ) {
-        $lib =~ m/-l(.*_(\w+)-.*)/ or die $lib;
+        $lib =~ m/-l(.*_(\w+))/ or die $lib;
         my( $key, $name ) = ( $2, $1 );
         $key = 'base' if $key =~ m/^base[ud]{0,2}/;
         $key = 'base' if $key =~ m/^carbon|^cocoa/ && $name !~ /osx_/; # here for Mac
@@ -34,7 +33,11 @@ sub awx_wx_config_data {
                               && $self->awx_is_monolithic
                               && $lib =~ m/(?:gtk2?|mac)[ud]{0,2}-/;
         my $dll = "lib${name}." . $self->awx_dlext;
-
+	my @t = glob("${LOCALBASE}/lib/$dll.*");
+	if ($#t >= 0) {
+		$dll = $t[-1];
+		$dll =~ s/.*\///g;
+	}
         $data{dlls}{$key} = { dll  => $dll,
                               link => $lib };
     }
